src/contract/abi.h

```diff
@@ -2,263 +2,306 @@
 #define ABI_H
 
 #include <string>
-
+#include <any>
 #include "../utils/hex.h"
 #include "../utils/json.hpp"
 #include "../utils/utils.h"
 
 /// Namespace for Solidity ABI-related operations.
 namespace ABI {
+/**
+ * Enum for the types of Solidity variables.
+ * Equivalency is as follows:
+ * - uint256 = uint256 (Solidity) = uint256_t (C++)
+ * - uint256Arr = uint256[] (Solidity) = std::vector<uint256_t> (C++)
+ * - address = address (Solidity) = Address (C++)
+ * - addressArr = address[] (Solidity) = std::vector<Address> (C++)
+ * - bool = bool (Solidity) = bool (C++)
+ * - boolArr = bool[] (Solidity) = vector<bool> (C++)
+ * - bytes = bytes (Solidity) = std::string (C++)
+ * - bytesArr = bytes[] (Solidity) = std::vector<std::string> (C++)
+ * - string = string (Solidity) = std::string (C++)
+ * - stringArr = string[] (Solidity) = std::vector<std::string> (C++)
+ */
+enum Types {
+  uint256,
+  uint256Arr,
+  address,
+  addressArr,
+  boolean,
+  booleanArr,
+  bytes,
+  bytesArr,
+  string,
+  stringArr
+};
+
+/// Class that encodes and packs native data types into Solidity ABI strings.
+class Encoder {
+private:
+  std::string data; ///< Encoded Solidity ABI string, as RAW BYTES. Use
```

Comment:
Doxygen might not understand the comment line split like that IIRC. You can either leave it all in one contiguous line:

```c++
std::string data; ///< Encoded Solidity ABI string, as RAW BYTES. Use Hex::fromBytes().get() to print it properly.
```

Or bring the comment to above the line:

```c++
/**
 * Encoded Solidity ABI string, as RAW BYTES.
 * Use Hex::fromBytes().get() to print it properly.
 */
std::string data;
```

Your call, but I suggest the latter since it's two lines anyway.

Reply:
Done!